### PR TITLE
julia: add mentoring notes to atbash-cipher

### PR DESCRIPTION
**Pull request by cmcaine**

### CODE REVIEW

File: tracks/julia/exercises/atbash-cipher/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    else
+        join(result)
+    end
+end
+
+function atbash(c::Char)
+    isletter(c) || return c
+    'z' - (lowercase(c) - 'a')
+end
+```
```
Comment (cmcaine) [inline tracks/julia/exercises/atbash-cipher/mentoring.md:100]:
`````suggestion
```
````
`````

File: tracks/julia/exercises/atbash-cipher/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+- Instead of writing out the whole alphabet, you can use a character range \``'a':'z'`\`, if you like.
+- \``Iterators.partition`\` and `join` might be helpful for adding spaces between groups.
+- Think about what happens to non-ascii input. What would you like to happen?
+
+The reverse index into an array is given by
+
+```julia
+length(arr) - idx
+```
```
Comment (SaschaMann) [inline tracks/julia/exercises/atbash-cipher/mentoring.md:11]: `Iterators.reverse` or `Base.reverseind` are other options.

**Review by SaschaMann — APPROVED** [review]
LGTM.

I'd find some notes on the (dis-)advantages of the example solutions useful as it's often not trivial to figure those out but this is obviously good to be merged as is :)

**Comment by cmcaine** [comment]
Updated with two new solutions adapted by me from others and with explanations. Also expanded the sample to include a nice data-oriented solution by Nosferican

**Review by SaschaMann — APPROVED** [review]
This is fantastic, thank you!

File: tracks/julia/exercises/atbash-cipher/mentoring.md [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+Compared to the more procedural approach above, this allocates more and performs extra work by creating various intermediate results that need to be transformed.
+
+The code uses `missing` because unencodable values do go missing and because `skipmissing` is a handy built-in iterator.
+I could also have used `nothing` and provided a function like `skipnothing(xs) = (x for x in xs if !isnothing(x))`.
+
+````
+By cmcaine, inspired by
+[halfdan](https://exercism.io/tracks/julia/exercises/atbash-cipher/solutions/419b6f4d04974a63b7f8531e8ad2808c):
+[OTDE](https://exercism.io/tracks/julia/exercises/atbash-cipher/solutions/330cb9e444c2403c898275e71da07caa)'s
+solutions.
```
Comment (cmcaine) [inline tracks/julia/exercises/atbash-cipher/mentoring.md:105]: Lol, I missed a word here.

**Pull request merged**